### PR TITLE
Make `sync_all_commands` errors easier to understand

### PR DESCRIPTION
**Pull request by LordOfPolls**
## About this pull request

Convert the indexes representing commands into command names

## Changes

- Wrapped `sync_all_commands`'s push into a try-except 
- Parse the discord error message for command indexes
- Determine which command corresponds to which index

## Checklist

- [x] I've run the `pre_push.py` script to format and lint code.
- [x] I've checked this pull request runs on `Python 3.6.X`.
- [ ] This fixes something in [Issues](https://github.com/eunwoo1104/discord-py-slash-command/issues).
    - Issue:
- [x] This adds something new.
- [ ] There is/are breaking change(s).
- [ ] (If required) Relevant documentation has been updated/added.
- [ ] This is not a code change. (README, docs, etc.)

### CODE REVIEW

File: discord_slash/client.py [hunk 1]
```diff
@@ -410,9 +411,30 @@ async def sync_all_commands(
...
+                            error_string = error_string.replace(
+                                f"In {num}",
+                                f"'{error_command.get('name')}'",
+                            )
+
+                        ex.args = (error_string,)
+                        raise ex
+
+                    else:
+                        raise ex
```
Comment (benwoo1110) [inline discord_slash/client.py]:
```suggestion

                    raise ex
```

Going to raise an exception regardless, the else part seems a bit redundant?

File: discord_slash/client.py [hunk 2]
```diff
@@ -410,9 +411,30 @@ async def sync_all_commands(
...
-                )
+                try:
+                    existing_cmds = await self.req.put_slash_commands(
+                        slash_commands=to_send, guild_id=scope
+                    )
+                except discord.HTTPException as ex:
+                    if ex.status == 400:
+                        # catch bad requests
+                        cmd_nums = set(
+                            re.findall(r"In\s(\d).", ex.args[0])
```
Comment (benwoo1110) [inline discord_slash/client.py:422]: What do you think about using `re.compile` it beforehand?

Comment (LordOfPolls) [reply discord_slash/client.py:422]: Due to how rarely this code should execute I didn't feel it was necessary.

No harm in adding it, but also very little benefit

File: discord_slash/client.py (hunk 1 above)
Comment (LordOfPolls) [reply discord_slash/client.py]: Very true. I'll rectify that

File: discord_slash/client.py (hunk 2 above)
Comment (benwoo1110) [reply discord_slash/client.py:422]: true, its your call lol

Comment (LordOfPolls) [reply discord_slash/client.py:422]: I'll add it. I guess theoretically someone could have hundreds of commands and that could take a while, relatively, to process

Comment (LordOfPolls) [reply discord_slash/client.py:422]: This code executes once per sync, on error. In which case it stops execution and raises an error. It only runs the regex once. So I won't add compile to it

(putting this here to explain my logic if it comes up later)

Comment (benwoo1110) [reply discord_slash/client.py:422]: Yea I get what you mean, the regex is only used if their program is bound to not work anyways.

**Review by benwoo1110 — APPROVED** [review]
looks good!

**Comment by LordOfPolls** [comment]
Requesting one more review

**Review by FayeDel — APPROVED** [review]
Tested, LGTM.

**Review by LilSpazJoekp — APPROVED** [review]
LGTM

**Review by i0bs — APPROVED** [review]

**Pull request merged**